colin-api/colin_api/models/filing.py

```diff
@@ -434,7 +394,53 @@ def _update_office(cls, cursor, event_id, corp_num, delivery_addr_id, mailing_ad
             raise err
 
     @classmethod
-    def find_ar(cls, business: Business = None, identifier: str = None, year: int = None):
+    def _find_filing_event_info(cls, identifier, filing_type_cd):
+
+        # build base querystring
+        querystring = (
+            """
+            select event.event_id, event_timestmp, first_nme, middle_nme, last_nme, email_addr
+            from event
+            join filing on filing.event_id = event.event_id
+            left join filing_user on event.event_id = filing_user.event_id
+            where filing_typ_cd=:filing_type_cd and corp_num=:identifier 
+            ORDER BY event_timestmp desc 
+            """
+        )
+
+        try:
+            cursor = db.connection.cursor()
+            cursor.execute(querystring, identifier=identifier, filing_type_cd=filing_type_cd)
+            event_info = cursor.fetchone()
+
+            if not event_info:
+                raise FilingNotFoundException(identifier=identifier, filing_type=filing_type_cd)
+
+            event_info = dict(zip([x[0].lower() for x in cursor.description], event_info))
+
+            # build filing user name from first, middle, last name
+            filing_user_name = ' '.join(
+                filter(None, [event_info['first_nme'], event_info['middle_nme'], event_info['last_nme']]))
+            filing_email = event_info['email_addr']
+
+            if not filing_user_name:
+                filing_user_name = 'N/A'
+
+                # if email is blank, set as empty tring
+            if not filing_email:
+                filing_email = 'N/A'
```

Comment:
must be in email format xxxx@xxxx.xxxx. I tested this theory by changing some test data in the "schemas" project.